### PR TITLE
chore: migrate navigation bar tests to TypeScript

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Related to #48753

I still need assistance with using rest operators in this instance.

### CODE REVIEW

**Comment by a2937** [comment]
(I meant for this to be a draft)

**Comment by a2937** [comment]
I'm still a little stumped because now I get illegal invocation errors.

**Comment by a2937** [comment]
The error

**Comment by a2937** [comment]
I believe that `win.EventTarget.prototype.addEventListener` is maybe a false positive for the following rule.

https://typescript-eslint.io/rules/unbound-method/

This function reference exists in the global scope and is not called until later with the apply method. And I think with the apply method; it scopes itself over there. What are your thoughts?

**Comment by raisedadead** [comment]
If you are struggling with something, we invite you to hit us in our contributor's chat room and forum. Chances are someone there will be able to guide you with your contribution more closely.

As for marking a PR draft, you can always toggle it in the side-panel when you are working on something :)

**Comment by Sboonny** [comment]
I have been following this PR for a while, and if I understood their question correctly, then it's a valid question that should be answered by the dev team, not the prolific contributors.

The whole section is anti-pattern of cypress best-practise [don't assign return values](https://docs.cypress.io/guides/references/best-practices#Assigning-Return-Values), and we should decided if we should keep it and use normal cypress to await for the page to load. or keep it for whatever reason we have, they don't know if we add this test for important reason.

```ts
let appHasStarted: boolean;
function spyOnListener(win: Cypress.AUTWindow, type: string, callback: EventListenerOrEventListenerObject | null, options?: boolean | AddEventListenerOptions | undefined) {

  const addEventListener = win.EventTarget.prototype.addEventListener;
  win.EventTarget.prototype.addEventListener = function (name: string) {
    if (name === 'click') {
      appHasStarted = true;
      win.EventTarget.prototype.addEventListener = addEventListener;
    }
    return addEventListener.apply(this, [type, callback, options]);
  };
}

function waitForAppStart() {
  return new Promise(resolve => {
    const isReady = () => {
      if (appHasStarted) {
        return resolve(null);
      }
      return setTimeout(isReady, 0);
    };
    isReady();
  });
}
```

So as dev-team, do we allow them to refactor this code, or not because there is an important reason for it?

**Comment by raisedadead** [comment]
> then it's a valid question that should be answered by the dev team, not the prolific contributors.

@Sboonny 

Just so that is not misinterpreted by anyone reading comments:

- Yes – the "who" in this particular query is _PROBABLY_ (not necessarily) the dev team. They are contributors who just happen to have a staff role. 

  All of our contributors are just incredible collaborators with the plethora of experience they bring, especially around best practices. 🙏🏽 

- The "where" for the queries is always the chat rooms and forum. They see way more activity than the PR regarding general coding help. 

  Whenever you get interest in your question, you can bring the conversation back to the PR/Issues for organizational purposes. 

  Also, the dev team hangs out everywhere :)

Cheers.

**Comment by a2937** [comment]
I am getting somewhere at least. 

Latest progress

**Comment by a2937** [comment]
I believe everything is ready for review now.

File: cypress/e2e/default/learn/common-components/navbar.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  navigationLinks: '.nav-list',
+  avatarContainer: '.avatar-container',
+  defaultAvatar: '.avatar-container',
+  menuButton: '.toggle-button-nav',
+  avatarImage: '.avatar-container .avatar'
+};
+
+
+
+
```
Comment (naomi-lgbt) [inline cypress/e2e/default/learn/common-components/navbar.ts:13]:
```suggestion

```

File: cypress/e2e/default/learn/common-components/navbar.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    cy.visit('/').then(() => cy.viewport(1200, 660));
+  });
+
+  it('Should have a "Sign in" button', () => {
+    cy.contains("[data-test-label='landing-small-cta']", 'Sign in');
+  });
+
+  it(
+    'Should have `Sign in` link on landing and learn pages' +
+    ' when not signed in',
```
Comment (naomi-lgbt) [inline cypress/e2e/default/learn/common-components/navbar.ts:25]:
```suggestion
    'Should have `Sign in` link on landing and learn pages' +
      ' when not signed in',
```

File: cypress/e2e/default/learn/common-components/navbar.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      cy.get(navBarselectors.menuButton).click();
+      cy.get(navBarselectors.navigationLinks).contains('Curriculum').click();
+      cy.contains(navBarselectors.smallCallToAction, 'Sign in');
+    }
+  );
+});
+
+describe('Navbar Logged in', () => {
+  beforeEach(() => {
+    cy.visit('/').then(() => cy.login()).then(() => cy.viewport(1200, 660));
```
Comment (naomi-lgbt) [inline cypress/e2e/default/learn/common-components/navbar.ts:37]:
```suggestion
    cy.visit('/')
      .then(() => cy.login())
      .then(() => cy.viewport(1200, 660));
```

File: cypress/e2e/default/learn/common-components/navbar.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        cy.get('svg').click();
+      });
+      cy.url().should('include', '/learn');
+    }
+  );
+
+  // have the curriculum and CTA on landing and /learn pages.
+  it(
+    'Should have `Radio`, `Forum`, and `Curriculum` links on landing and learn pages' +
+    'page when not signed in',
```
Comment (naomi-lgbt) [inline cypress/e2e/default/learn/common-components/navbar.ts:58]:
```suggestion
    'Should have `Radio`, `Forum`, and `Curriculum` links on landing and learn pages' +
      'page when not signed in',
```

File: cypress/e2e/default/learn/common-components/navbar.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  );
+
+  it('Should have `Profile` link when user is signed in', () => {
+    cy.get(navBarselectors.menuButton).click();
+    cy.get(navBarselectors.navigationLinks).contains('Profile').click();
+    cy.url().should('include', '/developmentuser');
+  });
+
+  it('Should have a profile image with class `default-border`', () => {
+    cy.get(navBarselectors.avatarContainer).should('have.class', 'default-border');
```
Comment (naomi-lgbt) [inline cypress/e2e/default/learn/common-components/navbar.ts:77]:
```suggestion
    cy.get(navBarselectors.avatarContainer).should(
      'have.class',
      'default-border'
    );
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
This should make the prettier step happy.

File: cypress/e2e/default/learn/common-components/navbar.ts (hunk 1 above)
Comment (a2937) [reply cypress/e2e/default/learn/common-components/navbar.ts]: That step was completed two days ago but I never managed to push it.

File: cypress/e2e/default/learn/common-components/navbar.ts (hunk 2 above)
Comment (a2937) [reply cypress/e2e/default/learn/common-components/navbar.ts]: That step was completed two days ago but I never managed to push it.

File: cypress/e2e/default/learn/common-components/navbar.ts (hunk 5 above)
Comment (a2937) [reply cypress/e2e/default/learn/common-components/navbar.ts]: That step was completed two days ago but I never managed to push it.

File: cypress/e2e/default/learn/common-components/navbar.ts (hunk 3 above)
Comment (a2937) [reply cypress/e2e/default/learn/common-components/navbar.ts]: That step was completed two days ago but I never managed to push it.

File: cypress/e2e/default/learn/common-components/navbar.ts (hunk 2 above)
Comment (a2937) [reply cypress/e2e/default/learn/common-components/navbar.ts]: That step was completed two days ago but I never managed to push it.

File: cypress/e2e/default/learn/common-components/navbar.ts (hunk 4 above)
Comment (a2937) [reply cypress/e2e/default/learn/common-components/navbar.ts]: That step was completed two days ago but I never managed to push it.

**Review by raisedadead — APPROVED** [review]

**Review by naomi-lgbt — APPROVED** [review]

**Comment by naomi-lgbt** [comment]
Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 🎉

**Pull request merged**